Public/Src/Cache/ContentStore/Distributed/NuCache/RocksDbContentLocationDatabase.cs

```diff
@@ -545,7 +549,16 @@ public override GetContentHashListResult GetContentHashList(OperationContext con
                         {
                             if (store.TryGetValue(key, out var data, nameof(Columns.Metadata)))
                             {
-                                result = DeserializeContentHashList(data);
+                                var metadata = DeserializeMetadataEntry(data);
+                                result = metadata.ContentHashListWithDeterminism;
+
+                                // Update the time, only if no one else has changed it in the mean time. We don't
+                                // really care if this succeeds or not, because if it doesn't it only means someone
+                                // else changed the stored value before this operation but after it was read.
+#pragma warning disable EPC13 // Suspiciously unobserved result.
+                                CompareExchange(context, strongFingerprint, metadata.ContentHashListWithDeterminism, metadata.ContentHashListWithDeterminism);

```

Comment:
>CompareExchange [](start = 32, length = 15)

I would say the selector -> last access time mapping should be a separate table.

Reply:
I don't think this is a good idea. There isn't any case in which we're interested on having one and not the other as far as I know, and it would either force two database roundtrips, which impacts performance, or somehow adapting IBuildXLKeyValueStore and all of that to work with WriteBatches

---
In reply to: [305424738](https://github.com/microsoft/BuildXL/pull/621#discussion_r305424738) [](ancestors = 305424738)